src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeStartUpTests.cs

```diff
@@ -0,0 +1,215 @@
+
+            await testSubject.Start();
+
+            testSubject.Process.HasExited.Should().BeFalse();
+
+            testSubject.Dispose();
+
+            testSubject.Process.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Dispose_ProcessNotStarted_NoException()
+        {
+            var testSubject = CreateTestSubject();
+
+            Action act = () => testSubject.Dispose();
+            act.Should().NotThrow();
+
+            testSubject.Process.Should().BeNull();
+        }
+
+        [TestMethod]
+        public async Task Dispose_ProcessAlreadyKilled_NoException()
+        {
+            var scriptPath = CreateScriptThatPrintsPortNumber(123, isHanging: true);
+            var startupScriptPath = " "; // pass whitespace so it would not fail NullOrEmpty validation but also do nothing
+
+            var nodeLocator = SetupNodeLocator(scriptPath);
+            var testSubject = CreateTestSubject(startupScriptPath, nodeLocator: nodeLocator.Object);
+
+            await testSubject.Start();
+            testSubject.Process.HasExited.Should().BeFalse();
+
+            testSubject.Dispose();
+            testSubject.Process.Should().BeNull();
+
+            Action act = () => testSubject.Dispose();
+            act.Should().NotThrow();
+
+            testSubject.Process.Should().BeNull();
+        }
+
+        private static Mock<INodeLocator> SetupNodeLocator(string nodePath)
+        {
+            var nodeLocator = new Mock<INodeLocator>();
+            nodeLocator.Setup(x => x.Locate()).Returns(nodePath);
+
+            return nodeLocator;
+        }
+
+        private EslintBridgeStartUp CreateTestSubject(string startupScriptPath = null, INodeLocator nodeLocator = null, IFileSystem fileSystem = null, ILogger logger = null)
+        {
+            startupScriptPath ??= "somefile.txt";
+            nodeLocator ??= SetupNodeLocator("some path").Object;
+            fileSystem ??= SetupStartupScriptFile(startupScriptPath, true).Object;
+            logger ??= Mock.Of<ILogger>();
+
+            return new EslintBridgeStartUp(startupScriptPath, nodeLocator, fileSystem, logger);
+        }
+
+        private static Mock<IFileSystem> SetupStartupScriptFile(string startupScriptPath, bool exists)
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(startupScriptPath)).Returns(exists);
+
+            return fileSystem;
+        }
+
+        private string CreateScriptThatPrintsPortNumber(int portNumber, bool isHanging = false)
+        {
+            var content = "echo port " + portNumber;
+
+            if (isHanging)
+            {
+                content += Environment.NewLine + "set /p asd=\"Waiting for process to be killed\"";
+            }
+
+            var fileName = Path.GetTempFileName() + ".bat";
+
+            File.WriteAllText(fileName, content);
```

Comment:
FYI originally I was deleting this file at the end of the test, but it was causing the tests to fail from time to time due to the file still being in use by another process. It could be the antivirus on my machine (like the problem I had with the cfamily subprocess). So in the end I decided to just leave the temp files and let the operating system clean them... Not the best solution, I know, but it's the same thing that we do in `ProcessRunnerTests` and I don't have another idea.

Reply:
I don't think it's worth worrying about. The files are tiny and will get cleaned up eventually.